TicTacToeApp/ViewController.swift

```diff
@@ -67,15 +105,27 @@ public class ViewController: UIViewController {
     alertController.addAction(UIAlertAction(title: title, style: .Default, handler: action))
   }
 
-  private func updateGameInformationDisplay() {
+  private func updateGamePrompt() {
     infoLabel.text = gamePrompt.promptFor(game)
   }
 
-  private func updateGameBoardButtonForPlayerMove(button: UIButton, playerMark: PlayerMark) {
+  private func updateGameBoardForPlayerMove(spaceId: Int, playerMark: PlayerMark) {
+    let button = getButtonByTag(spaceId) ?? UIButton()
+
     button.setTitle(getButtonTextFor(playerMark), forState: UIControlState.Normal)
     button.enabled = false
   }
 
+  private func getButtonByTag(spaceId: Int) -> UIButton? {
+    for button in boardButtons {
```

Comment:
Yeah, it doesn't strike me as particularly elegant, but this one-liner works:

return boardButtons.filter{ $0.tag == spaceId }.first